lib/client.js

```diff
@@ -553,6 +563,29 @@ Client.prototype.waitForSchemaAgreement = function (connection, callback) {
 };
 
 /**
+ * Waits for schema agreements and schedules schema metadata refresh.
+ * @param {Connection} connection
+ * @param event
+ * @param {Function} callback
+ * @ignore
+ * @internal
+ */
+Client.prototype.handleSchemaChange = function (connection, event, callback) {
+  var self = this;
+  this._waitForSchemaAgreement(connection, function (err) {
+    if (err) {
+      //we issue a warning but we continue with the normal flow
+      self.log('warning', 'There was an error while waiting for the schema agreement between nodes', err);
+    }
+    if (self.options.isMetadataEnabled) {
+      // schedule metadata refresh but do not wait for it to finish (no callback)
+      self.controlConnection.handleSchemaChange(event);
```

Comment:
Was just looking at the java driver to see what it does in this case (query response type is a schema change). It **does** wait for the updated schema to be fetched, but it is debounced, which delays the response by the debouncer interval ([JAVA-1120](https://datastax-oss.atlassian.net/browse/JAVA-1120)).  In this case the debouncing isn't necessary because the client was the one making the change, and it's very likely/good practice to make schema changes from one client.  With debouncing it would delay each schema change by 1 second, which is not ideal if making a bunch of schema changes, so I think it would be nice if we put that detection in here as well so it does not debounce if the client was the one that made the change.